In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
df_covid = pd.read_csv('covid19_italy_region.csv')

In [4]:
df_covid.head()

SNo                 Date Country  RegionCode      RegionName   Latitude  \
0    0  2020-02-24T18:00:00     ITA          13         Abruzzo  42.351222   
1    1  2020-02-24T18:00:00     ITA          17      Basilicata  40.639471   
2    2  2020-02-24T18:00:00     ITA          18        Calabria  38.905976   
3    3  2020-02-24T18:00:00     ITA          15        Campania  40.839566   
4    4  2020-02-24T18:00:00     ITA           8  Emilia-Romagna  44.494367   

   Longitude  HospitalizedPatients  IntensiveCarePatients  \
0  13.398438                     0                      0   
1  15.805148                     0                      0   
2  16.594402                     0                      0   
3  14.250850                     0                      0   
4  11.341721                    10                      2   

   TotalHospitalizedPatients  HomeConfinement  CurrentPositiveCases  \
0                          0                0                     0   
1                          0                0                     0   
2                          0                0                     0   
3                          0                0                     0   
4                         12                6                    18   

   NewPositiveCases  Recovered  Deaths  TotalPositiveCases  TestsPerformed  
0                 0          0       0                   0             NaN  
1                 0          0       0                   0             NaN  
2                 0          0       0                   0             NaN  
3                 0          0       0                   0             NaN  
4                18          0       0                  18             NaN

In [5]:
df_covid.describe()

SNo   RegionCode     Latitude    Longitude  \
count  6027.000000  6027.000000  6027.000000  6027.000000   
mean   3013.000000    11.857143    43.046293    12.225955   
std    1739.989368     6.274319     2.488987     2.659168   
min       0.000000     1.000000    38.115697     7.320149   
25%    1506.500000     7.000000    41.125596    11.121231   
50%    3013.000000    12.000000    43.616760    12.388247   
75%    4519.500000    17.000000    45.434905    13.768136   
max    6026.000000    22.000000    46.499335    16.867367   

       HospitalizedPatients  IntensiveCarePatients  TotalHospitalizedPatients  \
count           6027.000000            6027.000000                6027.000000   
mean             509.093579              55.892318                 564.985897   
std             1272.731157             134.957979                1403.362341   
min                0.000000               0.000000                   0.000000   
25%               16.000000               1.000000                  17.000000   
50%               93.000000               9.000000                 106.000000   
75%              384.500000              46.000000                 434.000000   
max            12077.000000            1381.000000               13328.000000   

       HomeConfinement  CurrentPositiveCases  NewPositiveCases      Recovered  \
count      6027.000000           6027.000000       6027.000000    6027.000000   
mean       5722.911399           6287.897296        286.885847    9337.852829   
std       15892.768789          16804.884786        818.450458   20551.265039   
min           0.000000              0.000000       -229.000000       0.000000   
25%         139.000000            168.000000          4.000000     734.000000   
50%         743.000000            905.000000         31.000000    2840.000000   
75%        3057.000000           3735.000000        159.500000    8450.500000   
max      155066.000000         164406.000000      11489.000000  289706.000000   

             Deaths  TotalPositiveCases  TestsPerformed  
count   6027.000000          6027.00000    4.872000e+03  
mean    1474.370665         17100.12361    2.600810e+05  
std     3365.190225         36934.67627    3.339818e+05  
min        0.000000             0.00000    3.482000e+03  
25%       98.000000          1383.50000    6.507375e+04  
50%      389.000000          4613.00000    1.365560e+05  
75%     1102.500000         15304.00000    3.043542e+05  
max    23024.000000        429109.00000    2.415099e+06

There is a high deviation from the average number of cases. This will be investigated further.

In [6]:
df_covid.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6027 entries, 0 to 6026
Data columns (total 17 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   SNo                        6027 non-null   int64  
 1   Date                       6027 non-null   object 
 2   Country                    6027 non-null   object 
 3   RegionCode                 6027 non-null   int64  
 4   RegionName                 6027 non-null   object 
 5   Latitude                   6027 non-null   float64
 6   Longitude                  6027 non-null   float64
 7   HospitalizedPatients       6027 non-null   int64  
 8   IntensiveCarePatients      6027 non-null   int64  
 9   TotalHospitalizedPatients  6027 non-null   int64  
 10  HomeConfinement            6027 non-null   int64  
 11  CurrentPositiveCases       6027 non-null   int64  
 12  NewPositiveCases           6027 non-null   int64  
 13  Recovered                  6027 non-null   int64

In [7]:
df_covid.shape

(6027, 17)

In [8]:
df_covid['NewPositiveCases'].sum()

1729061

In [9]:
df_covid['CurrentPositiveCases'].sum()

37897157

In [10]:
df_covid['TotalPositiveCases'].sum() == (df_covid['NewPositiveCases'].sum() + df_covid['CurrentPositiveCases'].sum())

False

In [11]:
df_covid['TotalPositiveCases'].sum()

103062445

In [12]:
df_covid.at[5000, 'NewPositiveCases']

108

In [13]:
df_covid.at[5000, 'Date']

'2020-10-19T17:00:00'

In [14]:
df_covid.loc[5000]

SNo                                         5000
Date                         2020-10-19T17:00:00
Country                                      ITA
RegionCode                                    18
RegionName                              Calabria
Latitude                               38.905976
Longitude                              16.594402
HospitalizedPatients                          66
IntensiveCarePatients                          5
TotalHospitalizedPatients                     71
HomeConfinement                             1099
CurrentPositiveCases                        1170
NewPositiveCases                             108
Recovered                                   1595
Deaths                                       104
TotalPositiveCases                          2869
TestsPerformed                          238117.0
Name: 5000, dtype: object

This returns the number of new cases of covid at row 5000 on October 19th, 2020 is 108.

In [15]:
df_covid['NewPositiveCases'].first_valid_index()

0

This checks the valid entry different from 'NaN'

### What is the total number of reported cases and reported death?

In [16]:
total_cases = df_covid.TotalPositiveCases.sum()
print(f"The total number of reported cases is {total_cases}")

The total number of reported cases is 103062445


In [17]:
total_death = df_covid.Deaths.sum()
print(f"The total number of death reported is {total_death}")

The total number of death reported is 8886032


It can be argued that not all reported cases of Covid19 resulted to the death recorded; that is, Covid is not the only contributing factor to death.

In [18]:
total_cases - total_death

94176413

### What is the death rate?

In [19]:
death_rate = total_death / total_cases
print(f"The ratio of death recorded to the number of cases is {death_rate}")

The ratio of death recorded to the number of cases is 0.08621988348908276


### What is the positivity rate?

In [21]:
positive_rate = (total_cases / df_covid.TestsPerformed.sum()) * 100
positive_rate

8.133631286870228

### Converting the date type to a datetime

In [22]:
df_covid['Date'] = pd.to_datetime(df_covid['Date'])

In [23]:
df_covid.Date

0      2020-02-24 18:00:00
1      2020-02-24 18:00:00
2      2020-02-24 18:00:00
3      2020-02-24 18:00:00
4      2020-02-24 18:00:00
               ...        
6022   2020-12-06 17:00:00
6023   2020-12-06 17:00:00
6024   2020-12-06 17:00:00
6025   2020-12-06 17:00:00
6026   2020-12-06 17:00:00
Name: Date, Length: 6027, dtype: datetime64[ns]

In [24]:
df_covid['year'] = pd.DatetimeIndex(df_covid.Date).year
df_covid['month'] = pd.DatetimeIndex(df_covid.Date).month
df_covid['day'] = pd.DatetimeIndex(df_covid.Date).day
df_covid['weekday'] = pd.DatetimeIndex(df_covid.Date).weekday

In [29]:
df_month = df_covid.groupby('month')[['NewPositiveCases', 'TestsPerformed', 'Deaths']].sum()
df_month

NewPositiveCases  TestsPerformed   Deaths
month                                           
2                  1120             0.0       96
3                104664             0.0   116566
4                 99671      13643342.0   644171
5                 27556      58918148.0   973585
6                  7584      86135311.0  1028841
7                  7006     113693227.0  1084927
8                 21702     140356556.0  1095146
9                 45623     181469958.0  1069879
10               364607     250655965.0  1137835
11               922206     342444409.0  1385098
12               127322      79797874.0   349888

### Death Cases in December

In [40]:
df_month.loc[12]

NewPositiveCases      127322.0
TestsPerformed      79797874.0
Deaths                349888.0
Name: 12, dtype: float64

### calculating the number of Tests per day

5901     906
5902     153
5903     301
5904    1721
5905    5805
        ... 
6022    1759
6023    2867
6024     460
6025     333
6026    4210
Name: Deaths, Length: 126, dtype: int64


### Visualization Of Data